x-pack/plugins/fleet/server/errors/handlers.ts

```diff
@@ -47,45 +54,72 @@ interface IngestErrorHandlerParams {
 }
 // unsure if this is correct. would prefer to use something "official"
 // this type is based on BadRequest values observed while debugging https://github.com/elastic/kibana/issues/75862
-
 const getHTTPResponseCode = (error: FleetError): number => {
-  if (error instanceof RegistryResponseError) {
-    // 4xx/5xx's from EPR
-    return 500;
+  // Bad Request
+  if (error instanceof PackageFailedVerificationError) {
+    return 400;
   }
-  if (error instanceof RegistryConnectionError || error instanceof RegistryError) {
-    // Connection errors (ie. RegistryConnectionError) / fallback  (RegistryError) from EPR
-    return 502; // Bad Gateway
+  if (error instanceof PackageOutdatedError) {
+    return 400;
   }
-  if (error instanceof PackageNotFoundError || error instanceof PackagePolicyNotFoundError) {
-    return 404; // Not Found
+  if (error instanceof PackageInvalidArchiveError) {
+    return 400;
   }
-  if (error instanceof AgentPolicyNameExistsError) {
-    return 409; // Conflict
+  if (error instanceof PackageRemovalError) {
+    return 400;
   }
-  if (error instanceof PackageUnsupportedMediaTypeError) {
-    return 415; // Unsupported Media Type
+  if (error instanceof KibanaSOReferenceError) {
+    return 400;
   }
-  if (error instanceof PackageFailedVerificationError) {
-    return 400; // Bad Request
+  // Unauthorized
+  if (error instanceof FleetUnauthorizedError) {
+    return 403;
   }
-  if (error instanceof ConcurrentInstallOperationError) {
-    return 409; // Conflict
+  // Not Found
+  if (error instanceof PackageNotFoundError || error instanceof PackagePolicyNotFoundError) {
+    return 404;
   }
   if (error instanceof AgentNotFoundError) {
     return 404;
   }
   if (error instanceof AgentActionNotFoundError) {
     return 404;
   }
-  if (error instanceof FleetUnauthorizedError) {
-    return 403; // Unauthorized
+  // Conflict
+  if (error instanceof AgentPolicyNameExistsError) {
+    return 409;
+  }
+  if (error instanceof ConcurrentInstallOperationError) {
+    return 409;
   }
   if (error instanceof PackagePolicyNameExistsError) {
-    return 409; // Conflict
+    return 409;
+  }
+  if (error instanceof PackageAlreadyInstalledError) {
+    return 409;
+  }
+  // Unsupported Media Type
+  if (error instanceof PackageUnsupportedMediaTypeError) {
+    return 415;
   }
+  // Internal Server Error
   if (error instanceof UninstallTokenError) {
-    return 500; // Internal Error
+    return 500;
+  }
+  if (error instanceof BundledPackageLocationNotFoundError) {
+    return 500;
+  }
+  if (error instanceof PackageESError) {
+    return 500;
+  }
+  if (error instanceof RegistryResponseError) {
+    // 4xx/5xx's from EPR
+    return 500;
```

Comment:
Does it also include 404s from EPR? Should we also return 404 on this case?

Reply:
I checked this one and it does never return 500 directly, it's just used to check when the registry responds a 500: https://github.com/search?q=repo%3Aelastic%2Fkibana+err+instanceof+RegistryResponseError+path%3A%2F%5Ex-pack%5C%2Fplugins%5C%2Ffleet%5C%2Fserver%5C%2F%2F&type=code

The actual response for a connection error from EPR seems to be this one (RegistryConnectionError): https://github.com/elastic/kibana/blob/21fc4cc3079db00d9898ec510049df03309b3faa/x-pack/plugins/fleet/server/services/epm/registry/requests.ts#L58-L60 and returns a 502